lib/annotations.go

```diff
@@ -0,0 +1,105 @@
+package metadata
+
+const (
+	// CheckpointAnnotationEngine is used by Container Checkpoint when
+	// creating a checkpoint image to specify the name of the container engine.
+	CheckpointAnnotationEngine = "org.criu.checkpoint.engine.name"
+
+	// CheckpointAnnotationEngineVersion is used by Container Checkpoint when
+	// creating a checkpoint image to specify the version of the container engine.
+	CheckpointAnnotationEngineVersion = "org.criu.checkpoint.engine.version"
+
+	// CheckpointAnnotationName is used by Container Checkpoint when
+	// creating a checkpoint image to specify the name of the checkpoint.
+	CheckpointAnnotationName = "org.criu.checkpoint.name"
+
+	// CheckpointAnnotationPod is used by Container Checkpoint when
+	// creating a checkpoint image to specify the name of the pod associated
+	// with the checkpoint.
+	CheckpointAnnotationPod = "org.criu.checkpoint.pod.name"
+
+	// CheckpointAnnotationPodID is used by Container Checkpoint when
+	// creating a checkpoint image to specify the ID of the pod associated
+	// with the checkpoint.
+	CheckpointAnnotationPodID = "org.criu.checkpoint.pod.id"
+
+	// CheckpointAnnotationNamespace is used by Container Checkpoint when
+	// creating a checkpoint image to specify the namespace of the pod associated
+	// with the checkpoint.
+	CheckpointAnnotationNamespace = "org.criu.checkpoint.pod.namespace"
+
+	// CheckpointAnnotationRootfsImage is used by Container Checkpoint when
+	// creating a checkpoint image to specify the root filesystem image associated
+	// with the checkpoint.
+	CheckpointAnnotationRootfsImage = "org.criu.checkpoint.rootfsImage"
```

Comment:
What is the purpose of `org.criu.checkpoint.rootfsImage` and how is it different from `org.criu.checkpoint.rootfsImageName`?

Reply:
Example -
```
"checkpointctl.annotation.checkpoint.rootfsImage": "docker.io/library/nginx@sha256:52478f8cd6a142fd462f0a7614a7bb064e969a4c083648235d6943c786df8cc7",
               "checkpointctl.annotation.checkpoint.rootfsImageID": "92b11f67642b62bbb98e7e49169c346b30e20cd3c1c034d31087e46924b9312e",
               "checkpointctl.annotation.checkpoint.rootfsImageName": "docker.io/library/nginx:latest",
```
I think `org.criu.checkpoint.rootfsImage`  containing some kind of hashed value in it, maybe calculated by some kind of cryptographic function like SHA256 for uniquely identify an image.

Reply:
@adrianreber I noticed that you have introduced `rootfsImage` (https://github.com/cri-o/cri-o/commit/38774e14a4f9e552cb01ccf71544eed8039e1da6) to make sure that we pull exactly the same base image during restore, and `rootfsImageName` seems to be redundant/irrelevant (https://github.com/cri-o/cri-o/commit/1eae23ee64548a90642e605ba920946970a71e2d). Would it make sense to remove `rootfsImageName` from the annotations here?

Reply:
The whole base rootfs image is currently not clear in different container engines from the checkpoint view.

We basically need three things. The image name the user has specified, the image name (including sha) which is actually used and probably the sha of the current image. In CRI-O this information is currently difficult to get, but should soon be available from what I have been told. So we need something like this:

- rootfsImageName (registry/path/image@sha:123456)
- rootfsImageUserRequested (registry/path/image:tag)
- rootfsImageSha or rootfsImageRef (123456)

The last one is not 100% necessary, but instead of parsing rootfsImageName to extract the sha it is easier to have it additionally listed.

Reply:
> rootfsImageUserRequested (registry/path/image:tag)

@adrianreber I don't understand the significance of the term 'UserRequested' here. Could you break it down for me a little?

Reply:
There is a function in CRI-O called [UserRequestedImage](https://github.com/cri-o/cri-o/blob/63cfc3b92ce26dfc6bc068529065d3c0687ce827/internal/oci/container.go#L360C21-L360C39):

>  UserRequestedImage returns the users' input originally used to find imageID; it might evaluate to a different image (or to a different kind of reference!) at any future time.